src/openrct2/actions/RideDemolishAction.hpp

```diff
@@ -340,7 +340,7 @@ DEFINE_GAME_ACTION(RideDemolishAction, GAME_COMMAND_DEMOLISH_RIDE, GameActionRes
             for (Direction dir : ALL_DIRECTIONS)
             {
                 const CoordsXY& off = DirOffsets[dir];
-                money32 removePrice = MazeRemoveTrack(location.x + off.x, location.y + off.y, location.z, dir);
+                money32 removePrice = MazeRemoveTrack({ location + off, dir });
```

Comment:
location + off results in a CoordsXY, with dir being interpreted as a Z.

You can prevent that by changing line 342 as follows:
```const CoordsXYZ& off = { DirOffsets[dir], 0 };```